components/service/glean/docs/metrics/adding-new-metrics.md

```diff
@@ -343,19 +343,34 @@ fun onShowLogin(e: Event) {
 }
 
 fun onLogin(e: Event) {
-    Auth.loginTime.stop(e.target)
+    Auth.loginTime.stopAndSum(e.target)
     // ...
 }
 
 fun onLoginCancel(e: Event) {
-    Auth.loginTime.cancel(e.tart)
+    Auth.loginTime.cancel(e.target)
     // ...
 }
 ```
 
 The time reported in the telemetry ping will be the sum of all of these
 timespans recorded during the lifetime of the ping.
 
+If the above `start`/`stopAndSum`/`cancel` API isn't practical to use from your
+application, you may use the alternative lower-level API `setRawNanos`. Unlike
+the above, this sets the value of the timespan directly, and does not sum it to
+the existing value, so use with caution if the span of the ping the metric is
+sent in may include multiple timings.
```

Comment:
This difference in behavior between the two ways to use the APIs is confusing.
We should follow up soon to have one consistent behavior here ([see this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1556798)) or make it consistent in this PR.

In general, if we have to call out inconsistent behavior like that, for me that's a flag for "does the API or design need changing?".

Reply:
As explained in the bug, I **strongly** think it would be wrong to change the existing API.  I'd be fine with changing `setRawNanos` to `addRawNanos` to be consistent.  I also think documenting that `setRawNanos` is only really safe if the ping is sent immediately after recording the metric would be a good idea.

Reply:
I'm fine with the additions here, the APIs are now convergent. @mdboom can you expand a bit on why `setRawNanos` is only safe if the ping is sent after recording? Is it because adding could change the value? If so, I'd rather phrase it differently, like "when 'setting' make sure no other operation takes places that modifies the value of the metric" :) From the usage seen so far, I suspect that's not too much of a big risk, but I agree it's worth documenting.

Reply:
Maybe using the word "safe" is confusing.  What I mean is -- if you use `setRawNanos` and then `setRawNanos` again before sending, that first measurement is totally gone.  Maybe someone would intend to do that, but it seems more likely that that would be a bug.  If you `setRawNanos` and then send the ping right away, you avoid that problem.  (It looks like the sync telemetry work is likely to do that, so at least in that instance it's doing the right thing).  I'm also fine to remove the "To be safe..." sentence altogether if it just adds confusion rather than clarifies.

Reply:
Oh, I see. Yes, I'd rather remove the "To be safe.." then.

Reply:
For this issue, changing the API is a risk, so please treat my points as "follow-up discussion".
We have an active discussion on the semantics on bugzilla, we can continue there.